### PR TITLE
#168 Ps update schema for conversations

### PR DESCRIPTION
**Pull request by AlexDjangoX**
ps-update-schema-for-conversations

Added models to Schema - Conversation, Message, UsersAndConversations

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -149,3 +151,34 @@ model DeliveryLogLine {
...
+  content           String
+  userId            Int
+  user              User          @relation(fields: [userId], references: [id])
+  conversationId    Int
+  conversation      Conversation  @relation(fields: [conversationId], references: [id])
+  createdAt         DateTime      @default(now())
+  updatedAt         DateTime      @updatedAt
+}
+
+model UsersAndConversations {
```
Comment (dearshrewdwit) [inline prisma/schema.prisma:177]: In the plan, this join table was going to be handled implicitly by Prisma.

Currently it's an explicit many:many, with a different name than 'ConversationMember'.

What's the reasoning here?

File: prisma/schema.prisma [hunk 2]
```diff
@@ -149,3 +151,34 @@ model DeliveryLogLine {
...
+}
+
+model Message {
+  id                Int           @id @default(autoincrement())
+  createdBy         String
+  content           String
+  userId            Int
+  user              User          @relation(fields: [userId], references: [id])
+  conversationId    Int
+  conversation      Conversation  @relation(fields: [conversationId], references: [id])
```
Comment (dearshrewdwit) [inline prisma/schema.prisma:172]: Let's cascade a delete on the conversation - if a conversation is deleted, so too should the corresponding messages.

File: prisma/schema.prisma [hunk 3]
```diff
@@ -149,3 +151,34 @@ model DeliveryLogLine {
...
+  createdAt         DateTime      @default(now())
+  updatedAt         DateTime      @updatedAt
+}
+
+model UsersAndConversations {
+userId            Int
+user              User  @relation(fields: [userId], references: [id])
+conversationId    Int
+conversation      Conversation  @relation(fields: [conversationId], references: [id])
```
Comment (dearshrewdwit) [inline prisma/schema.prisma:181]: If you're handling it explicitly, cascade the delete here too - if a conversation is deleted, so too should records of the conversation members

**Review by dearshrewdwit — CHANGES_REQUESTED** [review]

File: prisma/schema.prisma (hunk 1 above)
Comment (Tibor22) [reply prisma/schema.prisma:177]: We did this way so we can use query to do some filtering.

**Review by dearshrewdwit — CHANGES_REQUESTED** [review]
The schema migration is different to the one in #191 

Please choose one and update this or the other PR to match.

File: prisma/schema.prisma (hunk 1 above)
Comment (dearshrewdwit) [reply prisma/schema.prisma:158]: Ok - remember, the team process is to update the plan and get it reviewed by a teacher :)

**Review by dearshrewdwit — APPROVED** [review]
Normally I'd ask you to consolidate the migrations into one migration - there's no need to create tables and then drop and add columns. 

Locally, the workflow would have been to
- delete the first migration file
- reset the schema back to where it was. 
- generate the prisma client
- update the schema accordingly
- generate the prisma client
- generate the new migration to hold all changes.

**Pull request merged**